addons/l10n_ch/models/account_payment.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+from odoo import models
+
+class PaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    def _prepare_communication(self, invoices):
+        """Return a single ISR reference
+        to avoid duplicate of the same number when multiple payments are done
+        on the same reference. As those payments are grouped by reference,
+        we want a unique reference in communication.
+
+        """
+        if invoices[0]._is_isr_supplier_invoice():
```

Comment:
This is a little weird, since you're actually depending on the function below to do that... At least, put some comment explaining that.

Reply:
Maybe the docstring of the method is not enough? How do you want me to improve it?

Maybe I should add an example?

In short we want that:

`120000000000234478943216899 120000000000234478943216899`
be
`120000000000234478943216899`

This is tested in the unit tests.

Reply:
So this is only for communications. As shown in the example, we want to avoid having a repetition of the same structured ref that would break the generation of SEPA file with structured ref.

Reply:
I added a comment in the code.